slack-api-client/src/main/java/com/slack/api/rate_limits/metrics/impl/BaseMemoryMetricsDatastore.java

```diff
@@ -314,9 +366,30 @@ public void run() {
                     continue;
                 }
                 String executorName = executor.getKey().replaceFirst("^" + type + "/", "");
+
+                if (this.store.isTraceMode()) {
+                    log.debug("Going to maintain {} metrics (executor: {}, teams: {})",
+                            this.store.getMetricsType(),
+                            executorName,
+                            executor.getValue().keySet().size()
+                    );
+                }
                 for (ConcurrentMap.Entry<String, LiveRequestStats> team : executor.getValue().entrySet()) {
                     String teamId = team.getKey();
                     LiveRequestStats stats = team.getValue();
+                    if (stats == null) {
+                        continue;
+                    }
+                    if (stats.getLastRequestTimestampMillis() != null
+                            && stats.getLastRequestTimestampMillis() <= this.lastExecutionTimestampMillis) {
```

Comment:
If there is no new request since the previous `run()` method execution, we can safely skip the following operations here.